tests/playbooks/container_distribution.yaml

```diff
@@ -0,0 +1,143 @@
+---
+- hosts: localhost
+  gather_facts: false
+  vars_files:
+    - vars/server.yaml
+  module_defaults: &pulp_module_defaults
+    pulp.squeezer.container_distribution: &pulp_connection_details
+      pulp_url: "{{ pulp_url }}"
+      username: "{{ pulp_username }}"
+      password: "{{ pulp_password }}"
+      validate_certs: "{{ pulp_validate_certs | default(true) }}"
+    pulp.squeezer.container_remote:
+      <<: *pulp_connection_details
+    pulp.squeezer.container_sync:
+      <<: *pulp_connection_details
+    pulp.squeezer.container_repository:
+      <<: *pulp_connection_details
+    pulp.squeezer.x509_cert_guard:
+      <<: *pulp_connection_details
+  tasks:
+    - name: Make repository absent
+      pulp.squeezer.container_repository:
+        name: test_container_repository
+        state: absent
+    - name: Make distribution absent
+      pulp.squeezer.container_distribution:
+        name: test_container_distribution
+        state: absent
+    - name: Make repository present
+      pulp.squeezer.container_repository:
+        name: test_container_repository
+        state: present
+    - name: Make remote present
+      pulp.squeezer.container_remote:
+        name: test_container_remote
+        upstream_name: pulp/pulp-fixtures
+        url: "https://registry-1.docker.io"
+        state: present
+    - name: Sync remote into repository
+      pulp.squeezer.container_sync:
+        remote: test_container_remote
+        repository: test_container_repository
+    - name: Clean openapi cache
+      file:
+        path: "{{ lookup('env', 'XDG_CACHE_HOME') | default('~/.cache') }}/squeezer"
+        state: absent
+
+- hosts: tests
+  gather_facts: false
+  vars_files:
+    - vars/server.yaml
+  module_defaults:
+    <<: *pulp_module_defaults
+  tasks:
+    - name: Retrieve repository
+      pulp.squeezer.container_repository:
+        name: test_container_repository
+      register: repository_result
+
+    - name: Distribute repository
+      pulp.squeezer.container_distribution:
+        name: test_container_distribution
+        base_path: test_container_base_path
+        repository: test_container_repository
+        state: present
+      register: result
+    - name: Verify distribution of repository
+      assert:
+        that:
+          - result.changed == true
+          - result.distribution.name == "test_container_distribution"
+          - result.distribution.base_path == "test_container_base_path"
+          - ansible_check_mode or result.distribution.registry_path is regex(".*/test_container_base_path")
+          - result.distribution.repository == repository_result.repository.pulp_href
+          - ansible_check_mode or result.distribution.content_guard is regex("/pulp/api/v3/contentguards/*")
```

Comment:
Maybe it's been too long and my brain wiped some corner cases, but why should these not work in check_mode?

Reply:
The other fields can be determined from the module arguments, but these are only returned by Pulp once the distribution is created.

Reply:
Right. `registry_path` is not `base_path`, i agree. But with respect to the content_guard, i disagree. See the corresponding `file_distribution` test.

https://github.com/pulp/squeezer/blob/develop/tests/playbooks/file_distribution.yaml#L86

Reply:
It failed without it. This content guard is different, since the redirect gets added automatically.

Reply:
Ah, I should have know that. :facepalm: